### PR TITLE
User: support the created_at attribute

### PR DESCRIPTION
**Pull request by bgarciaentornos**

### CODE REVIEW

**Comment by buckett** [comment]
I think it's better to use `Instant` rather than `Date` as we're only supporting back to Java 8.
It is possible to also add support for `updated_at` at the same time, just for completeness?

**Comment by bgarciaentornos** [comment]
Hi Matthew. Initially I tried both things but it seems there's no `updated_at` field for users and the date conversion was giving me error with the data I was receiving. I'll try to improve the conversion and update this.

**Comment by bgarciaentornos** [comment]
Had to modify the GsonResponseParser, not sure if any integrations will expect the previous format?

**Comment by buckett** [comment]
Could you look to see if there are any other classes that use `Instant` that should be deserialised from JSON and check to see if they have tests.

**Comment by bgarciaentornos** [comment]
Another example would be the CalendarEvent startAt / endAt fields, which already have some existing tests. But all the existing unit tests are working and I added another one using this new serialization.

**Comment by buckett** [comment]
Ok, have you checked to see if CalendarEvent is the ***only*** other place where Instants are used?

**Comment by bgarciaentornos** [comment]
No, there are more places and even more of Date cases for similar fields too, both with or without unit tests.

**Comment by buckett** [comment]
Ok, but it's `Instant`s that you're changing the parser for as I understand it. Are there any other uses of `Instant`?

**Comment by bgarciaentornos** [comment]
CalendarEvent (x4), ContentMigration (x3), File (x5), SisImport (x4). Most of them don't have a unit test for that field.

**Comment by buckett** [comment]
Ok, so my concern here is that how can we be confident that this change isn't going to break anything in any of those areas?

**Comment by bgarciaentornos** [comment]
Yep, that was my initial concern and the reason why I left it as a Date, as other recent additions. I wasn't sure if this specific case should override the existing (for 3 years) parsing, but I'm also wondering if it was actually used at all considering the parsing was failing with the format currently returned by Canvas.

What do you think I should do then?

**Comment by buckett** [comment]
Why not support both formats in the parsing?
Or add tests for the objects that currently don't?

**Comment by bgarciaentornos** [comment]
Ok, I can do both but that'll have to wait then. Looking at previous PRs I thought it wasn't necessary, but it is if we're changing the parser.

**Comment by buckett** [comment]
Ok, if we have tests that cover the existing uses I'm fine with changing the parser behaviour, otherwise it seems risky (don't you think?), supporting both means I'm less worried about breaking existing code.

This is 2020 and we should be writing new code that uses `Date`.

**Comment by bgarciaentornos** [comment]
Agree, I'll try to work on the tests tomorrow or Monday.

**Comment by bgarciaentornos** [comment]
I had to add a test class for sis imports, didn't cover every method or field though.

File: src/test/resources/SampleJson/contentMigration/GetContentMigrationCompleted.json [hunk 1]
```diff
@@ -2,10 +2,10 @@
 	"id": 44,
 	"user_id": 899123456,
 	"workflow_state": "completed",
-	"started_at": "2020-06-09T09:26:30Z",
-	"finished_at": "2020-06-09T09:26:52Z",
+	"started_at": "2020-06-09T09:26:30-06:00",
+	"finished_at": "2020-06-09T09:26:52-06:00",
 	"migration_type": "course_copy_importer",
-	"created_at": "2020-06-09T09:26:30Z",
+	"created_at": "2020-06-09T09:26:30-06:00",
```
Comment (buckett) [inline src/test/resources/SampleJson/contentMigration/GetContentMigrationCompleted.json:8]: Was the original JSON not correct, or have Instructure changed how they are formatting the dates?

Comment (bgarciaentornos) [reply src/test/resources/SampleJson/contentMigration/GetContentMigrationCompleted.json:8]: I've seen it both ways (see some of the others json I've modified), but this new parser handles both formats, unlike the previous one.

Comment (buckett) [reply src/test/resources/SampleJson/contentMigration/GetContentMigrationCompleted.json:8]: But shouldn't the JSON in the tests be in the format that Canvas returns as that's what we want to tests?

Comment (bgarciaentornos) [reply src/test/resources/SampleJson/contentMigration/GetContentMigrationCompleted.json:8]: Yes, that's why I changed it here. You can see it on:

https://canvas.instructure.com/doc/api/content_migrations.html

But also as I said I saw some dates with the Z format on some other Canvas responses/documentation.

**Pull request merged**